src/services/chain.ts

```diff
@@ -189,7 +189,11 @@ export class ChainContext {
     let fromBlock = lastProcessedBlock
       ? lastProcessedBlock.number + 1
       : this.deploymentBlock;
+
     let currentBlock = await provider.getBlock("latest");
+    metrics.blockHeightLatest
+      .labels(chainId.toString())
+      .set(currentBlock.number);
```

Comment:
Nitpick: since this is repeated 3 times, would it be worth it to wrap around a fn? Just so we don't forget to track the metrics if we want to use this elsewhere.

Reply:
sorry, i saw it only after merging. The whole code needs some serious refactoring. Lets clean it up all in the future 🙏